utils/decypharr_settings.py

```
@@ -1,7 +1,403 @@
+    url: str,
+    key: str,
+    method: str = "GET",
+    data: Optional[dict] = None,
+    timeout: int = 10,
+):
+    headers = {"X-Api-Key": key, "Accept": "application/json"}
+    body = None
+    if data is not None:
+        headers["Content-Type"] = "application/json"
+        body = json.dumps(data).encode("utf-8")
+    req = urllib.request.Request(url, data=body, headers=headers, method=method)
```

Ruff (0.14.8)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)